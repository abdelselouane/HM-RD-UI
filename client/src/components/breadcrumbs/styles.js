import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
