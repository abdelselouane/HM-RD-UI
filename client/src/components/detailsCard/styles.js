import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
