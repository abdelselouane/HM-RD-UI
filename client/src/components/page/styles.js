import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
