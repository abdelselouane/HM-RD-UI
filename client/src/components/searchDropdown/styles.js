import styled from 'styled-components';

const StyledSearchDropdown = styled.div`
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);
  z-index: 5;

  .scrollable-table {
    overflow-y: hidden;
  }

  .standard-table tbody tr {
    td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &:hover td {
      background-color: ${({ theme }) => theme.black10} !important;
    }
  }

  .table-body.alternating-rows {
    width: 100%;
    
    tr:nth-child(2n + 1),
    tr:nth-child(2n) {
      background-color: ${({ theme }) => theme.black10};
    }

    tr:hover td,
    tr:focus td {
      background-color: ${({ theme }) => theme.black12} !important;
    }
    
    tr td {
      border: none;
    }
  }
`;

export default StyledSearchDropdown;
