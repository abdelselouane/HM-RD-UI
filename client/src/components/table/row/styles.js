import styled from 'styled-components';

const StyledTd = styled.td`
  ${props => {
    if (props.tdProps
      && props.tdProps.color
      && Object.keys(props.theme).includes(props.tdProps.color)) {
      return {
        ...props.tdProps,
        color: `${props.theme[props.tdProps.color]} !important`
      };
    }
    if (props.disableRow) {
      return {
        ...props.tdProps,
        color: `${props.theme.black14} !important`
      };
    }
    return props.tdProps;
  }}
`;

export default StyledTd;
