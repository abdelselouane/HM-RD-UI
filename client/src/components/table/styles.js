import styled, { css } from 'styled-components';

const expandableTableCss = css`
  &.standard-table > thead > tr > th {
    background-color: ${({ theme }) => theme.white} !important; 
    &.sortable {
      &:after {
        position: static !important;
        padding-left: 10px;
      }
    }  
  }

  &.standard-table > tbody > tr {
    > td {
      border-bottom: 1px solid ${({ theme }) => theme.black14};
    }

    &:nth-child(4n + 1):hover > td,
    &:nth-child(4n + 2):hover > td,
    &:nth-child(4n + 3):hover > td,
    &:nth-child(4n + 4):hover > td {
      background-color: unset !important;
    }

    &:nth-child(4n + 1) td,
    &:nth-child(4n + 2) td {
      background-color: ${({ theme }) => theme.black11} !important;
    }

    &:nth-child(4n + 3) td,
    &:nth-child(4n + 4) td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &[id^='expandable'] {
      &:nth-child(4n + 1):hover,
      &:nth-child(4n + 1):hover > td,
      &:nth-child(4n + 2):hover,
      &:nth-child(4n + 2):hover > td {
        background-color: ${({ theme }) => theme.black11} !important;
      }

      &:nth-child(4n + 3):hover,
      &:nth-child(4n + 3):hover > td,
      &:nth-child(4n + 4):hover,
      &:nth-child(4n + 4):hover > td {
        background-color: ${({ theme }) => theme.white} !important;
      }
    }
  }
`;

const clearOnHoverCss = css`
  &.standard-table > thead > tr:hover,
  &.standard-table > thead > tr:hover > th {
    background-color: inherit !important;
  }
`;

export const StyledTable = styled.table`
  background-color: unset !important;
  ${({ isExpandable }) => isExpandable && expandableTableCss} ${({ clearOnHover }) =>
  clearOnHover && clearOnHoverCss};
`;

export const StyledTh = styled.th`
  ${props => {
    if (
      props.thProps &&
      props.thProps.color &&
      Object.keys(props.theme).includes(props.thProps.color)
    ) {
      return {
        ...props.thProps,
        color: `${props.theme[props.thProps.color]} !important`
      };
    }
    return props.thProps;
  }};
`;
