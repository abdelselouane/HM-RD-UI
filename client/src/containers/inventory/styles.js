import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
