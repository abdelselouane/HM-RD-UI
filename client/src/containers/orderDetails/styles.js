import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
