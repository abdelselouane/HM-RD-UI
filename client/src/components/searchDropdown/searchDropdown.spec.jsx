/* global describe, document, beforeAll, it, expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import SearchDropdown from './index';
import * as Styles from './styles';

import Table from '../table';

describe('<SearchDropdown />', () => {
  let wrapper;
  let props;
  let mockElement;

  beforeAll(() => {
    React.createRef = jest.fn(() => ({
      current: {
        focus: jest.fn(),
        contains: jest.fn()
          .mockReturnValueOnce(false)
          .mockReturnValueOnce(true)
      }
    }));

    ReactDOM.createPortal = jest.fn();

    mockElement = {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      setAttribute: jest.fn()
    };

    document.querySelector = jest.fn(() => mockElement);

    props = {
      columns: [
        {
          key: 'mockData',
          name: 'Mock Data'
        }
      ],
      parentSelector: 'mockElementId',
      rows: [
        { mockData: 'mockValue' }
      ],
      searchKey: 'mockData',
      className: 'mock-classname',
      handleInputChange: jest.fn(),
      handleSelectClick: jest.fn()
    };

    wrapper = shallow(<SearchDropdown {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    const instance = wrapper.instance();
    expect(toJson(shallow(instance.renderDropdown()))).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
  });

  it('should call addEventListener on componentWillMount', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(mockElement, 'addEventListener');
    instance.componentWillMount();
    expect(spy).toHaveBeenCalledWith('mousedown', instance.handleCloseDropdown, false);
  });

  it('should call removeEventListener on componentWillUnmount', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(mockElement, 'removeEventListener');
    instance.componentWillUnmount();
    expect(spy).toHaveBeenCalledWith('mousedown', instance.handleCloseDropdown, false);
  });

  it('should set showDropdown to false in handleCloseDropdown if dropdownRef.current.contains(target) is true', () => {
    // React.createRef().current.contains.mockReturnValueOnce(false);
    wrapper.setState({ showDropdown: true });
    const instance = wrapper.instance();
    instance.handleCloseDropdown({ target: 'mock target' });
    expect(instance.state.showDropdown).toBe(false);
  });

  it('should not change showDropdown in handleCloseDropdown if dropdownRef.current.contains(target) is false', () => {
    // React.createRef().current.contains.mockReturnValueOnce(true);
    wrapper.setState({ showDropdown: true });
    const instance = wrapper.instance();
    instance.handleCloseDropdown({ target: 'mock target' });
    expect(instance.state.showDropdown).toBe(true);
  });

  it('should set showDropdown to true in handleFocus', () => {
    wrapper.setState({ showDropdown: false });
    const instance = wrapper.instance();
    instance.handleFocus();
    expect(instance.state.showDropdown).toBe(true);
  });

  it('should set showDropdown to false in handleRowClick', () => {
    wrapper.setState({ showDropdown: true });
    const instance = wrapper.instance();
    const row = instance.getOptions()[0];
    instance.handleRowClick(row);
    expect(instance.state.showDropdown).toBe(false);
  });

  it('should set showDropdown to false and call props.handleSelectClick if defined in handleRowClick', () => {
    const newProps = {
      ...props,
      handleSelectClick: jest.fn()
    };
    wrapper.setProps(newProps);
    const spy = jest.spyOn(newProps, 'handleSelectClick');
    const instance = wrapper.instance();
    const row = instance.getOptions()[0];
    instance.handleRowClick(row);
    expect(spy).toHaveBeenCalledWith({ key: 'mockValue', mockData: 'mockValue' });
  });

  it('should set searchValue in handleInputChange', () => {
    wrapper.setState({ searchValue: '' });
    const instance = wrapper.instance();
    instance.handleInputChange('searchValue', 'a');
    expect(instance.state.searchValue).toBe('a');
  });

  it('should call props.handleInputChange if defined in handleInputChange', () => {
    const newProps = {
      ...props,
      handleInputChange: jest.fn()
    };
    wrapper.setProps(newProps);
    const spy = jest.spyOn(newProps, 'handleInputChange');
    const instance = wrapper.instance();
    instance.handleInputChange('mockId', 'mockValue');
    expect(spy).toHaveBeenCalledWith('mockId', 'mockValue');
    spy.mockReset();
    spy.mockRestore();
  });

  it('should set state.portalElement after componentDidMount', () => {
    wrapper.setState({ portalElement: null });
    const instance = wrapper.instance();
    ReactDOM.createPortal.mockReturnValueOnce(instance.renderDropdown());
    const portalSpy = jest.spyOn(ReactDOM, 'createPortal');
    const elementSpy = jest.spyOn(mockElement, 'setAttribute');

    instance.componentDidMount();

    expect(elementSpy).toHaveBeenCalledWith('style', 'position: relative');
    expect(instance.state.portalElement).toBe(mockElement);
    expect(portalSpy).toHaveBeenCalledWith(instance.renderDropdown(), mockElement);
  });

  describe('getOptions', () => {
    it('should only encapsulate matching characters from searchValue and row[searchKey] in a strong anchor', () => {
      wrapper.setState({ searchValue: 'mock' });
      const instance = wrapper.instance();
      let optionSearchElement = instance.getOptions()[0][instance.props.searchKey];
      expect(shallow(optionSearchElement).find('strong').text()).toBe('mock');

      wrapper.setState({ searchValue: '' });
      optionSearchElement = instance.getOptions()[0][instance.props.searchKey];
      expect(shallow(optionSearchElement).find('strong').text()).toHaveLength(0);
    });
  });

  describe('renderDropdown', () => {
    it('should give table no options if showDropdown is false', () => {
      wrapper.setState({ showDropdown: false });
      const instance = wrapper.instance();
      const DropdownWrapper = shallow(instance.renderDropdown());
      expect(DropdownWrapper.find(Table).props().rows).toHaveLength(0);
    });
  });
});
