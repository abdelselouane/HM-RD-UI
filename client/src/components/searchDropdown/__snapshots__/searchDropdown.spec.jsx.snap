// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchDropdown /> should render properly 1`] = `
<Fragment>
  <TextInputField
    className="mock-classname dropdown input-field"
    containerRef={
      Object {
        "current": Object {
          "contains": [MockFunction],
          "focus": [MockFunction],
        },
      }
    }
    icon={
      Object {
        "className": "icon_search",
      }
    }
    id="mockData"
    onChange={[Function]}
    onFocus={[Function]}
    setFocusOnMount={true}
  />
</Fragment>
`;

exports[`<SearchDropdown /> should render properly 2`] = `
.c0 {
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  z-index: 5;
}

.c0 .scrollable-table {
  overflow-y: hidden;
}

.c0 .standard-table tbody tr td {
  background-color: !important;
}

.c0 .standard-table tbody tr:hover td {
  background-color: !important;
}

.c0 .table-body.alternating-rows {
  width: 100%;
}

.c0 .table-body.alternating-rows tr:hover td,
.c0 .table-body.alternating-rows tr:focus td {
  background-color: !important;
}

.c0 .table-body.alternating-rows tr td {
  border: none;
}

<div
  className="c0"
>
  <Table
    className=""
    clearOnHover={false}
    columns={
      Array [
        Object {
          "key": "mockData",
          "name": "Mock Data",
        },
      ]
    }
    handleInputChange={[MockFunction]}
    handleRowClick={[Function]}
    handleSelectClick={[MockFunction]}
    handleSortClick={[Function]}
    isAlternating={true}
    isExpandable={false}
    rows={
      Array [
        Object {
          "key": "mockValue",
          "mockData": <span
            title="mockValue"
          >
            <strong>
              
            </strong>
            mockValue
          </span>,
        },
      ]
    }
    showBody={true}
    showCheckbox={false}
    showHeader={false}
    showResults={false}
  />
</div>
`;

exports[`<SearchDropdown /> should render styles properly 1`] = `
.c0 {
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  z-index: 5;
}

.c0 .scrollable-table {
  overflow-y: hidden;
}

.c0 .standard-table tbody tr td {
  background-color: !important;
}

.c0 .standard-table tbody tr:hover td {
  background-color: !important;
}

.c0 .table-body.alternating-rows {
  width: 100%;
}

.c0 .table-body.alternating-rows tr:hover td,
.c0 .table-body.alternating-rows tr:focus td {
  background-color: !important;
}

.c0 .table-body.alternating-rows tr td {
  border: none;
}

<div
  className="c0"
/>
`;
